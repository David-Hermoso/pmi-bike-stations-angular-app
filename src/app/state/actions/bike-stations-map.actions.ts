import {createAction, props} from '@ngrx/store';
import {BikeStation} from '../../models/bike-station';

export const bikeStationsMapActionsIds = {
  showBikeStationInfo: '[Bike stations map] show bike station info',
  closeBikeStationInfo: '[Bike stations info] close bike station info'
};

export const showBikeStationInfo = createAction(bikeStationsMapActionsIds.showBikeStationInfo,
  props<{ payload: BikeStation }>());

export const closeBikeStationInfo = createAction(bikeStationsMapActionsIds.closeBikeStationInfo);
