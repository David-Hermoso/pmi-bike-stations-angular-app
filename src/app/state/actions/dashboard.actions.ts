import {createAction, props} from '@ngrx/store';
import {BikeStation} from '../../dashboard';

export const dashboardActionsIds = {
  getBikeStationsStart: '[Dashboard] getBikeStations start',
  getBikeStationsSuccess: '[Dashboard] getBikeStations success',
  getBikeStationsError: '[Dashboard] getBikeStations error'
};

export const getBikeStationsStart = createAction(dashboardActionsIds.getBikeStationsStart);
export const getBikeStationsSuccess = createAction(dashboardActionsIds.getBikeStationsSuccess,
  props<{ bikeStations: BikeStation[] }>());
export const getBikeStationsError = createAction(dashboardActionsIds.getBikeStationsError);
