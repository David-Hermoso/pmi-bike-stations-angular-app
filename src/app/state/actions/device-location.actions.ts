import {createAction, props} from '@ngrx/store';

export const deviceLocationActionsIds = {
  request: '[Device location] request',
  success: '[Device location] success',
  error: '[Device location] error',
};

export const deviceLocationRequest = createAction(deviceLocationActionsIds.request);

export const deviceLocationSuccess = createAction(deviceLocationActionsIds.success,
  props<{ deviceLocation: any }>());
export const deviceLocationError = createAction(deviceLocationActionsIds.error);
