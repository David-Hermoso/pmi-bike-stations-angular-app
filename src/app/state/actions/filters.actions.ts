import {createAction, props} from '@ngrx/store';
import {Filters} from '../../models/filters';

export const filtersActionsIds = {
  filter: '[Filters] filter',
};

export const filter = createAction(filtersActionsIds.filter,
  props<{ payload: Filters }>());
