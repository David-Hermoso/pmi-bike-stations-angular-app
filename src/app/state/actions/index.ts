export * from './bike-stations-list.actions';
