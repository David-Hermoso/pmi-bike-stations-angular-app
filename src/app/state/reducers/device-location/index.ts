export * from './device-location.reducer';
