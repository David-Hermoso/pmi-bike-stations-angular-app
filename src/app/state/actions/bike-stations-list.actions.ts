import {createAction, props} from '@ngrx/store';
import {BikeStationsState} from '../reducers';

export const bikeStationsGalleryActionsIds = {
  getBikeStationsStart: '[Bike stations gallery] getBikeStations start',
  getBikeStationsSuccess: '[Bike stations gallery] getBikeStations success',
  getBikeStationsError: '[Bike stations gallery] getBikeStations error'
};

export const getBikeStationsStart = createAction(bikeStationsGalleryActionsIds.getBikeStationsStart);
export const getBikeStationsSuccess = createAction(bikeStationsGalleryActionsIds.getBikeStationsSuccess,
  props<BikeStationsState>());
export const getBikeStationsError = createAction(bikeStationsGalleryActionsIds.getBikeStationsError);
