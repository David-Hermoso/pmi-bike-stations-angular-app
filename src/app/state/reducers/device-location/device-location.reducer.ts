import {createReducer, on} from '@ngrx/store';
import * as DeviceLocationActions from '../../actions/device-location.actions';
import {AppState} from '../../../models/app.state';

export const selectDeviceLocation = (appState: AppState) => appState.deviceLocation;

export const deviceLocationReducer = createReducer(
  undefined as unknown,
  on(DeviceLocationActions.deviceLocationSuccess,
    (currentState: any, {deviceLocation}) => deviceLocation
  )
);
