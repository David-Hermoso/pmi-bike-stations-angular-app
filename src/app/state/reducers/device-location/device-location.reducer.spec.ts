import * as fromDeviceLocation from './device-location.reducer';
import {deviceLocationSuccess} from '../../actions/device-location.actions';

describe('deviceLocationReducer', () => {
  describe('unknown action', () => {
    it('should return the default state', () => {
      const deviceLocation = {
        lat: 3.87,
        lng: 2.56
      };
      const action = {
        type: 'Unexistant action',
      };
      const state = fromDeviceLocation.deviceLocationReducer({
        deviceLocation,
      }, action);

      expect(state).toMatchSnapshot();
    });
  });
  it('should set selectedBikeStation', () => {
    const deviceLocation = {
      lat: 3.87,
      lng: 2.56
    };
    const action = deviceLocationSuccess({
      deviceLocation,
    });
    const state = fromDeviceLocation.deviceLocationReducer(undefined, action);

    expect(state).toMatchSnapshot();
  });
});
