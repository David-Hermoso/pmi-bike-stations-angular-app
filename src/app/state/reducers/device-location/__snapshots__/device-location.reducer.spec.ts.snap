// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deviceLocationReducer should set selectedBikeStation 1`] = `
Object {
  "lat": 3.87,
  "lng": 2.56,
}
`;

exports[`deviceLocationReducer unknown action should return the default state 1`] = `
Object {
  "deviceLocation": Object {
    "lat": 3.87,
    "lng": 2.56,
  },
}
`;
